File: src/types/domain.ts
export interface User {
  id: string;
  email: string;
  name?: string;
}

export interface Goal {
  id: string;
  userId: string;
  title: string;
  description?: string;
  targetDate?: string;
  status: 'active' | 'completed' | 'paused' | 'archived';
  priority: 'low' | 'medium' | 'high';
  createdAt: string;
  updatedAt: string;
}

export interface Roadmap {
  id: string;
  goalId: string;
  title: string;
  createdAt: string;
}

export interface Milestone {
  id: string;
  roadmapId: string;
  title: string;
  description?: string;
  status: 'todo' | 'completed';
  dueDate?: string;
}

export interface ScheduleBlock {
  id: string;
  userId: string;
  taskId?: string;
  habitId?: string;
  title: string;
  startTime: string;
  endTime: string;
  type: 'TASK' | 'EVENT' | 'FOCUS' | 'BREAK' | 'HABIT';
  reminderId?: string;
  reminderEnabled?: boolean;
}

export interface Habit {
  id: string;
  userId: string;
  title: string;
  frequency: 'DAILY' | 'WEEKLY';
  createdAt: string;
}

export interface HabitEntry {
  id: string;
  habitId: string;
  date: string;
  completed: boolean;
}

export interface Task {
  id: string;
  goalId?: string;
  title: string;
  description?: string;
  status: 'todo' | 'in_progress' | 'completed' | 'blocked';
  dueDate?: string;
  estimatedDuration?: number; // in minutes
  actualDuration?: number;
  priority: 'low' | 'medium' | 'high';
  createdAt: string;
  updatedAt: string;
}

File: server/src/main/kotlin/com/orbit/models/Domain.kt
package com.orbit.models

import kotlinx.serialization.Serializable

@Serializable
data class User(
    val id: String,
    val email: String,
    val name: String? = null,
    val passwordHash: String
)

@Serializable
data class Goal(
    val id: String,
    val userId: String,
    val title: String,
    val description: String? = null,
    val targetDate: String? = null,
    val status: GoalStatus = GoalStatus.ACTIVE,
    val priority: Priority = Priority.MEDIUM,
    val createdAt: String,
    val updatedAt: String
)

@Serializable
enum class GoalStatus {
    ACTIVE, COMPLETED, PAUSED, ARCHIVED
}

@Serializable
enum class Priority {
    LOW, MEDIUM, HIGH
}

@Serializable
data class Task(
    val id: String,
    val userId: String,
    val goalId: String? = null,
    val projectId: String? = null,
    val title: String,
    val description: String? = null,
    val status: TaskStatus = TaskStatus.TODO,
    val priority: Priority = Priority.MEDIUM,
    val dueDate: String? = null,
    val estimatedDuration: Int? = null,
    val actualDuration: Int? = null,
    val createdAt: String,
    val updatedAt: String
)

@Serializable
enum class TaskStatus {
    TODO, IN_PROGRESS, COMPLETED, BLOCKED
}

@Serializable
data class Roadmap(
    val id: String,
    val goalId: String,
    val title: String,
    val createdAt: String
)

@Serializable
data class Milestone(
    val id: String,
    val roadmapId: String,
    val title: String,
    val description: String? = null,
    val status: MilestoneStatus = MilestoneStatus.TODO,
    val dueDate: String? = null
)

@Serializable
enum class MilestoneStatus {
    TODO, COMPLETED
}

@Serializable
data class Project(
    val id: String,
    val userId: String,
    val goalId: String? = null,
    val title: String,
    val description: String? = null,
    val status: ProjectStatus = ProjectStatus.ACTIVE
)

@Serializable
enum class ProjectStatus {
    ACTIVE, COMPLETED, ARCHIVED
}

@Serializable
data class ScheduleBlock(
    val id: String,
    val userId: String,
    val taskId: String? = null,
    val title: String,
    val startTime: String,
    val endTime: String,
    val type: ScheduleBlockType = ScheduleBlockType.TASK
)

@Serializable
enum class ScheduleBlockType {
    TASK, EVENT, FOCUS, BREAK
}
